src/components/app/UrlManager.js

```diff
@@ -141,13 +141,24 @@ class UrlManager extends React.PureComponent<Props> {
       }
     }
 
+    // Profile sanitization and publishing can do weird things for the history API.
+    // Rather than write lots of complicated interactions, just prevent the back button
+    // from working when going between a published profile, and one that is not.
+    const isOneDataSourcePublic =
+      previousUrlState.dataSource === 'public' ||
+      newUrlState.dataSource === 'public';
+    const isOneDataSourceFromAddon =
+      previousUrlState.dataSource === 'from-addon' ||
+      newUrlState.dataSource === 'from-addon';
+    const doesDataSourceChange =
+      previousUrlState.dataSource !== newUrlState.dataSource;
+    const doesHashChange = previousUrlState.hash !== newUrlState.hash;
     if (
-      previousUrlState.dataSource !== newUrlState.dataSource ||
```

Comment:
Blocking: I can't really reason with these changes. Can you please provide an STR for what this is fixing? I'm a bit concerned with what is going on here. I would prefer this change to come after #2673, and include a test showing the breakage, with what is being fixed. It's hard to tell if we are mistakenly replacing state when we shouldn't be. I know this area is really error prone, so I would like to be careful here.

Reply:
A STR that wasn't working is:
* go on the compare page
* go to the home page by clicking in the new header
* press the <back> button

=> that wasn't working because of the existing code.

I realized that the existing code tried to prevent going back to "from-addon" after publishing a profile. So I tried to retain this behavior but still allow moving between "no-profiles" datasources, and also from `/public` or `/from-url` to home.

But now I think we should rather remove this code at all and use your code from #2673 to "replace state" when publishing, instead of "pushing state".

Reply:
For now I'm extracting this commit our of the PR.